tests/unit/test_charm.py

```diff
@@ -1769,47 +1771,55 @@ def test_on_get_password_secrets(harness):
         mock_event.set_results.assert_called_once_with({"password": "replication-test-password"})
 
 
-@pytest.mark.parametrize("scope", [("app"), ("unit")])
+@pytest.mark.parametrize("scope,field", [("app", "operator-password"), ("unit", "csr")])
 @patch_network_get(private_address="1.1.1.1")
-def test_get_secret_secrets(harness, scope):
+def test_get_secret_secrets(harness, scope, field):
     with (
         patch("charm.PostgresqlOperatorCharm._on_leader_elected"),
     ):
         harness.set_leader()
 
-        assert harness.charm.get_secret(scope, "operator-password") is None
-        harness.charm.set_secret(scope, "operator-password", "test-password")
-        assert harness.charm.get_secret(scope, "operator-password") == "test-password"
+        assert harness.charm.get_secret(scope, field) is None
+        harness.charm.set_secret(scope, field, "test")
+        assert harness.charm.get_secret(scope, field) == "test"
 
 
 @patch_network_get(private_address="1.1.1.1")
-def test_set_secret(harness):
+def test_set_secret(harness, _has_secrets):
+    # after new data_interfaces version, every secret we set using set_secret
+    # should not go to the databag, but to juju secrets instead.
+    if _has_secrets:
```

Comment:
In the past, before secrets were available, this test was intended to check relation data because the secrets were exactly there. IMHO, we can now update the test by using `get_secret` to check whether the operations worked correctly.

Reply:
If we change the `get_relation_data` in this test for `get_secret`, then it would become pretty much a copy of `test_set_reset_new_secret`, right? (setting with `set_secret` and asserting with `get_secret`). Do we want to do this?

Reply:
Actually, I think I got what you mean, we could use get_secret on the specific asserts using the relation data, now I see. Adapted the code, thanks!